rct_optimizations/include/rct_optimizations/serialization/eigen.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <Eigen/Geometry>
+#include <yaml-cpp/yaml.h>
+
+namespace YAML
+{
+template<typename V>
+struct convert<Eigen::Matrix<V, 2, 1>>
```

Comment:
Is there any possibility of these functions colliding with their equivalents in [`message_serialization`](https://github.com/swri-robotics/message_serialization/blob/master/include/message_serialization/eigen_yaml.h) if both are included in the same file? I made a very simple test case to check for this and it didn't immediately cause problems, but I don't know if there's some other subtle issue to watch out for.

Reply:
I'm not entirely sure; I haven't experienced any problems with it either yet. I think this is something we can figure out in the future if it happens though

Reply:
Why don't we just add these to message_serialization instead of here. These data types are not specific to robot calibration in any way.

Reply:
Most, if not all, of these functions are already in `message_serialization`. I avoided adding a dependency on `message_serialization` here because it is currently a ROS1 package and `rct_optimizations` is a pure CMake package. At this point I don't mind duplicating a small number of serialization functions if it means not adding an external build-from-source dependency.